universal-login-relayer/lib/core/services/messages/QueueService.ts

```diff
@@ -27,11 +28,14 @@ class QueueService {
   }
 
   async execute(messageHash: string) {
-    await this.messageRepository.setMessageState(messageHash, 'Pending');
     try {
       const signedMessage = await this.messageRepository.getMessage(messageHash);
-      const {hash} = await this.messageExecutor.executeAndWait(signedMessage);
-      await this.messageRepository.markAsSuccess(messageHash, hash!);
+      const transactionResponse = await this.messageExecutor.execute(signedMessage);
+      const {hash} = transactionResponse;
+      ensureNotNull(hash, TransactionHashNotFound);
+      await this.messageRepository.markAsPending(messageHash, hash!);
+      await this.messageExecutor.waitAndHandleTransaction(transactionResponse);
```

Comment:
We should move OnTransactionSent to Queue form executor and inline waiting to this method. No reason to push it down the abstract later.